tools/filter.py

```diff
@@ -0,0 +1,150 @@
+# This file is part of the EESSI build-and-deploy bot,
+# see https://github.com/EESSI/eessi-bot-software-layer
+#
+# The bot helps with requests to add software installations to the
+# EESSI software layer, see https://github.com/EESSI/software-layer
+#
+# author: Thomas Roeblitz (@trz42)
+#
+# license: GPLv2
+#
+import re
+
+from collections import namedtuple
+from pyghee.utils import log
+
+
+FILTER_COMPONENTS = ['architecture', 'instance', 'job', 'repository']
+
+Filter = namedtuple('Filter', ('component', 'pattern'))
+
+
+class EESSIBotActionFilterError(Exception):
+    pass
+
+
+class EESSIBotActionFilter:
+    def __init__(self, filter_string):
+        """
+        EESSIBotActionFilter constructor
+
+        Args:
+            filter_string (string): string containing whitespace separated filters
+        """
+        self.action_filters = []
+        for _filter in filter_string.split():
+            try:
+                self.add_filter_from_string(_filter)
+            except EESSIBotActionFilterError:
+                raise
+            except Exception as err:
+                log(f"Unexpected err={err}, type(err)={type(err)}")
+                raise
+
+    def clear_all(self):
+        self.action_filters = []
+
+    def add_filter(self, component, pattern):
+        """
+        Adds a filter
+
+        Args:
+            component (string): any prefix of 'architecture', 'instance', 'job' or 'repository'
+            pattern (string): regex that is applied to a string representing the component
+        """
+        # check if component is supported
+        full_component = None
+        for cis in FILTER_COMPONENTS:
+            if cis.startswith(component):
+                full_component = cis
+                break
+        if full_component:
+            log(f"processing component {component}")
+            # if full_component == architecture replace - with / in pattern
+            if full_component == 'architecture':
```

Comment:
`'architecture'` should be a constant, so above:

```python
FILTER_COMPONENT_ARCH = 'architecture'
FILTER_COMPONENTS = [FILTER_COMPONENT_ARCH, ...]

Reply:
ack